src/software/python_bindings/python_binding_utilities.cpp

```diff
@@ -0,0 +1,80 @@
+#include "software/python_bindings/python_binding_utilities.h"
+
+#include "software/util/variant_visitor/variant_visitor.h"
+
+namespace py = pybind11;
+
+template <typename ParamValueType>
+void setParameterValueFromDictIfExists(std::shared_ptr<Parameter<ParamValueType>> param,
+                                       const py::dict& config_update_dict)
+{
+    if (config_update_dict.contains(param->name()))
+    {
+        param->setValue(
+            config_update_dict[param->name().c_str()].template cast<ParamValueType>());
+    }
```

Comment:
maybe throw an exception or something here? I feel this could pretty easily lead to silent failures.

Reply:
The dict is intended to override the default parameters instead of specifying all of them, the only error case would be if we tried to update a parameter that doesn't exist. Added a check for this case.